From 6131ed03b3f3e66f26eb4b384fc1ab888db47f27 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Sat, 14 Dec 2013 18:53:20 +0000
Subject: [PATCH] builder: Only use virt-resize --no-sparse when writing to a
 block device.

Previously we would use the virt-resize --no-sparse option if the
final output file was a block device.  This is safe, but unnecessary
for example if virt-resize was used as an intermediate step.  So only
use this option if virt-resize is actually writing to the block
device.

(cherry picked from commit 0037820ee003bfbc71bf35b18fb3a066cbf298f8)
---
 builder/builder.ml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/builder/builder.ml b/builder/builder.ml
index 8951be5..148c9bf 100644
--- a/builder/builder.ml
+++ b/builder/builder.ml
@@ -493,7 +493,7 @@ let main () =
       let cmd =
         sprintf "virt-resize%s%s%s --output-format %s%s%s %s %s"
           (if debug then " --verbose" else " --quiet")
-          (if output_is_block_dev then " --no-sparse" else "")
+          (if is_block_device ofile then " --no-sparse" else "")
           (match iformat with
           | None -> ""
           | Some iformat -> sprintf " --format %s" (quote iformat))
-- 
1.8.5.3

